embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/metadata/HostedSdkVersionInfo.kt

```diff
@@ -0,0 +1,221 @@
+package io.embrace.android.embracesdk.capture.metadata
+
+import io.embrace.android.embracesdk.Embrace
+import io.embrace.android.embracesdk.logging.InternalEmbraceLogger
+import io.embrace.android.embracesdk.prefs.PreferencesService
+
+internal class HostedSdkVersionInfo(
+    private val preferencesService: PreferencesService,
+    private val logger: InternalEmbraceLogger,
+    appFramework: Embrace.AppFramework = Embrace.AppFramework.NATIVE
+) {
+    private var hostedPlatformStrategy: HostedPlatformStrategy
+
+    // precompute on initialization
+    init {
+        when (appFramework) {
+            Embrace.AppFramework.REACT_NATIVE -> this.hostedPlatformStrategy = ReactNativePlatformStrategy()
+            Embrace.AppFramework.UNITY -> this.hostedPlatformStrategy = UnityPlatformStrategy()
+            Embrace.AppFramework.FLUTTER -> this.hostedPlatformStrategy = FlutterPlatformStrategy()
+            else -> this.hostedPlatformStrategy = NativePlatformStrategy()
+        }
+    }
+    var hostedSdkVersion: String? = null
+        get() = field ?: hostedPlatformStrategy.getHostedSdkVersionFromPreferences(preferencesService)
+        set(value) {
+            val sdkVersion = hostedPlatformStrategy.getHostedSdkVersionFromPreferences(preferencesService)
+            if (sdkVersion != null) {
+                logger.logDeveloper("Embrace", "hostedSdkVersion is present")
```

Comment:
nit: it's probably ok to get rid of all these logDeveloper messages. We don't really use them anymore & our test coverage is much better than when we first added them in for debugging purposes.